client/components/store-details-page.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import Map from './map';
+
+export default class StoreDetailsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      details: {},
+      events: []
+    };
+    this.getAddress = this.getAddress.bind(this);
+    this.getStoreEvents = this.getStoreEvents.bind(this);
+  }
+
+  getAddress() {
+    fetch(`/api/address/?storeName=${this.props.match.params.name}`)
+      .then(data => data.json())
+      .then(result => result.results.filter(store => store.name === this.props.match.params.name))
+      .then(data => this.setState({ details: data[0] }))
+      .catch(err => console.error(err));
+  }
+
+  getStoreEvents() {
+    fetch(`/api/storeEvents/?storeName=${this.props.match.params.name}`)
```

Comment:
Right now, backend is params ('/api/storeEvents/:storeName') we can discuss if we want to change the backend or frontend for that endpoint tomorrow!